File: 0846-hand-of-straights/0846-hand-of-straights.java
class Solution {
    public boolean findSucessors(int[]hand,int groupSize, int i, int n){
        int target = hand[i]+1;
        hand[i] = -1;
        int count = 1;
        i += 1;
        while(i < n && count < groupSize){
            if(hand[i] == target){
                target = hand[i]+1;
                hand[i] = -1;
                count++;
            }
            i++;
        }
        return count == groupSize;
    }
    public boolean isNStraightHand(int[] hand, int groupSize) {
        int n = hand.length;
        if(n%groupSize != 0) return false;
        Arrays.sort(hand);
        for(int i = 0; i < n; i++){
            if(hand[i] >= 0 && !findSucessors(hand,groupSize,i,n)){
                return false;
            }
        }
        return true;
    }
}

File: 0846-hand-of-straights/0846-hand-of-straights.py
class Solution(object):
    def isNStraightHand(self, hand, groupSize):
        n = len(hand)
        if n % groupSize != 0:
            return False
        hand.sort(reverse = False)
        for i in range(0,n):
            if(hand[i] >= 0 and not self.findSucessors(hand,groupSize,i,n)):
                return False
        return True
    def findSucessors(self,hand,groupSize,i,n):
        target = hand[i] + 1
        hand[i] = -1
        count = 1
        i = i + 1
        while(count < groupSize and i < n):
            if target == hand[i]:
                target = hand[i] + 1
                hand[i] = -1
                count+=1
            i += 1
        return count == groupSize

        